java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -1076,22 +1127,26 @@ class DatabaseImpl(
         suspend fun insertFieldBlock(
             fieldName: String,
             fieldType: FieldType,
-            isCollection: Boolean
+            isCollection: FieldClass
         ) {
             counters?.increment(DatabaseCounters.INSERT_ENTITY_FIELD)
             insertFieldStatement.apply {
                 bindLong(1, getTypeId(fieldType, db))
                 bindLong(2, schemaTypeId)
                 bindString(3, fieldName)
-                bindBoolean(4, isCollection)
+                bindLong(4, isCollection.ordinal.toLong())
                 executeInsert()
             }
         }
         schema.fields.singletons.forEach { (fieldName, fieldType) ->
-            insertFieldBlock(fieldName, fieldType, isCollection = false)
+            val isCollection = when {
```

Comment:
instead of isCollection, call this fieldClass?
Also, you need to update the comment on the is_collection field in the create statement

Reply:
I can't change the name without a DB migration; I was thinking that I'd do the name change as part of a migration I'll need for BigInt. WDYT?

Reply:
I was just suggesting changing the name of the local variable (but it doesn't matter much). What matters is updating the comment in the schema.

Reply:
oh, you just mean here specifically.. done.